lib/sandbox/execute.js

```diff
@@ -231,6 +240,47 @@ module.exports = function (bridge, glob) {
             timers.clearEvent(id, err, res, history);
         });
 
+        bridge.on(runRequestResponseEventName, function (id, err, response, results) {
+            // Apply variable mutations from the scripts executed via pm.execution.runRequest
+            // to the current execution scope
+            const { variableMutationStages = {} } = results || {};
+
+            ['prerequest', 'test'].forEach(function (scriptStage) {
+                const variableMutationForThisStage = variableMutationStages[scriptStage];
+
+                if (!variableMutationForThisStage) {
+                    return;
+                }
+
+                ['_variables', 'environment', 'globals'].forEach(function (variableType) {
+                    let mutableVariableScope = execution[variableType],
+                        mutationsMade = variableMutationForThisStage[variableType];
+
+                    if (!mutationsMade) {
+                        return;
+                    }
+
+                    if (_.isFunction(mutableVariableScope.applyMutation)) {
+                        mutationsMade = new sdk.MutationTracker(mutationsMade);
+                        mutationsMade.applyOn(mutableVariableScope);
+                    }
+                });
+            });
+
+            timers.clearEvent(id, err, response);
+        });
+
+        // For a parallel execution to receive variable values from this execution
+        bridge.on(requestForVariablesEventName, function (requestUniqueId) {
```

Comment:
Instead of adding a new event, we can send the context (for now, just variables) out from the parent execution to the consumer via the execution.run_collection_request event [here](https://github.com/postmanlabs/postman-sandbox/pull/1094/files#diff-f7717fbd7530109af763f0596041cb93947808ebd2bda845ef1d544b79d7edf3R335)